Content.Client/SimpleStation14/Examine/CharacterInformation/UI/CharacterInformationWindow.xaml.cs

```diff
@@ -0,0 +1,163 @@
+
+        _name.SetMarkup(unknown);
+        _job.SetMarkup(unknown);
+
+        _flavor.SetMarkup("Placeholder flavor text.");
+    }
+
+    /// <summary>
+    ///     Updates the UI to show all relevant information about the entity
+    /// </summary>
+    /// <param name="examined">The entity to become informed about</param>
+    /// <param name="name">The name of the examined entity, taken from their ID</param>
+    /// <param name="job">The job of the examined entity, taken from their ID</param>
+    /// <param name="flavorText">The flavor text of the examined entity</param>
+    public void UpdateUi(EntityUid examined, string? name = null, string? job = null, string? flavorText = null)
+    {
+        ResetUi();
+
+        // Fill in the omnidirectional sprite views
+        if (_entity.TryGetComponent<SpriteComponent>(examined, out var sprite))
+            FillSprites(sprite, examined);
+
+        // Fill in the name and job
+        if (!string.IsNullOrEmpty(name))
+            _name.SetMarkup(name);
+        if (!string.IsNullOrEmpty(job))
+            _job.SetMarkup(job);
+
+        // Fill in the flavor text
+        if (!string.IsNullOrEmpty(flavorText))
+        {
+            _flavor.SetMessage(flavorText);
+            _rootWindow.MinSize = new Vector2(675, 384);
+            _rootWindow.SetSize = new Vector2(675, 384);
+            _separator.Visible = true;
+            _flavorTextScroll.Visible = true;
+        }
+        else
+        {
+            _rootWindow.MinSize = new Vector2(292, 384);
+            _rootWindow.SetSize = new Vector2(292, 384);
+            _separator.Visible = false;
+            _flavorTextScroll.Visible = false;
+        }
+    }
+
+
+    /// <summary>
+    ///     Fills the sprite views with the sprite from the sprite component
+    /// </summary>
+    /// <param name="sprite">Sprite component to use</param>
+    /// <param name="entity">The entity <paramref name="sprite"/> belongs to</param>
+    private void FillSprites(SpriteComponent sprite, EntityUid entity)
+    {
+        // This all should "freeze" the sprite views
+        // Spawn a dummy entity to get a copy of the sprite component from
+        _dummy = _entity.SpawnEntity(_entity.GetComponent<MetaDataComponent>(entity).EntityPrototype!.ID, MapCoordinates.Nullspace);
+
+        // Ensures the dummy has the same sex as the entity so masks are applied correctly
+        if (_entity.TryGetComponent(entity, out HumanoidAppearanceComponent? humanoidAppearance))
+        {
+            var newHumanoidAppearance = _entity.EnsureComponent<HumanoidAppearanceComponent>(_dummy);
+            newHumanoidAppearance.Sex = humanoidAppearance.Sex;
+        }
+
+        // Spawn and equip a fake jumpsuit (it won't be shown) so the appearance system doesn't destroy reality when applying masks
+        var clothing = _entity.SpawnEntity("ClothingUniformJumpsuitColorGrey", MapCoordinates.Nullspace);
```

Comment:
It doesn't currently matter for the version of Harpies present in this Repo, but this'll cause problems when you merge the 1.0 version of the Harpy species. For Harpies, Lamia, and Arachne, I've deprecated the AnytaurMask in favor of code enforced skirts, but this is predicated on them never being able to equip a jumpsuit. You'll want to grab the HumanoidCharacterProfile from entity, and then have this system spawn and equip a jumpskirt if the profile has skirt selected.

Reply:
It won't, it forces equipping.